lib/private/Files/Cache/Cache.php

```diff
@@ -123,7 +123,7 @@ public function __construct(IStorage $storage) {
 
 		$this->storageCache = new Storage($storage);
 		$this->mimetypeLoader = \OC::$server->getMimeTypeLoader();
-		$this->connection = \OC::$server->getDatabaseConnection();
+		$this->connection = \OC::$server->get(IDBConnection::class);
```

Comment:
```suggestion
		$this->connection = \OCP\Server::get(IDBConnection::class);
```

let's use the public server container API. same applies to most changes below

Reply:
I forgot to make this change before pushing, and I'm currently running tests. I don't know if you can cancel the tests that are currently running to free up resources, but feel free.

Also, why the change to the public container? In general, when should this be used?

Reply:
> In general, when should this be used?

In new code: always
In old code: when you touch it anyways